File: backend/Concord/Concord.Application/Models/MemberModerationResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class MemberModerationResponse
{
    [JsonPropertyName("UserId")]
    public Guid UserId { get; set; }

    [JsonPropertyName("IsMuted")]
    public bool IsMuted { get; set; }

    [JsonPropertyName("TimedOutUntil")]
    public DateTime? TimedOutUntil { get; set; }

    [JsonPropertyName("TimedOutByUserId")]
    public Guid? TimedOutByUserId { get; set; }

    [JsonPropertyName("TimeoutReason")]
    public string? TimeoutReason { get; set; }
}

File: mobile/concord/lib/api/models/member_moderation_response.dart
import 'json_utils.dart';

class MemberModerationResponse {
  const MemberModerationResponse({
    required this.userId,
    required this.isMuted,
    required this.timedOutUntil,
    required this.timedOutByUserId,
    required this.timeoutReason,
  });

  factory MemberModerationResponse.fromJson(Map<String, dynamic> json) {
    return MemberModerationResponse(
      userId: json.field('UserId') as String,
      isMuted: json.field('IsMuted') as bool,
      timedOutUntil: parseNullableDateTime(json.field('TimedOutUntil')),
      timedOutByUserId: json.field('TimedOutByUserId') as String?,
      timeoutReason: json.field('TimeoutReason') as String?,
    );
  }

  final String userId;
  final bool isMuted;
  final DateTime? timedOutUntil;
  final String? timedOutByUserId;
  final String? timeoutReason;
}
